app/components/primer/layout.rb

```diff
@@ -0,0 +1,347 @@
+    #   <% end %>
+    #   <%= render(Primer::Layout.new(flow_row_until: :md, mt: 5)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #   <%= render(Primer::Layout.new(flow_row_until: :lg, mt: 5)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #
+    # @example Density
+    #
+    #   <%= render(Primer::Layout.new(density: :none)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #   <%= render(Primer::Layout.new(density: :compact)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #   <%= render(Primer::Layout.new(density: :normal)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #   <%= render(Primer::Layout.new(density: :relaxed)) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #
+    # @example Three column layout
+    #
+    #   <%= render(Primer::Layout.new) do |c| %>
+    #     <% c.main(border: true) do %>
+    #       <%= render(Primer::Layout.new(sidebar_placement: :end)) do |l| %>
+    #         <% l.main(border: true) { "Main" } %>
+    #         <% l.sidebar(border: true) { "Metadata" } %>
+    #       <% end %>
+    #     <% end %>
+    #     <% c.sidebar(border: true) { "Sidebar" } %>
+    #   <% end %>
+    #
+    # @param density [Symbol] Margin around the `Layout`.
+    # @param container [Symbol] Container to wrap the `Layout` in. <%= one_of(Primer::Layout::CONTAINER_OPTIONS) %>
+    # @param gutter [Symbol] Space between `main` and `sidebar`. <%= one_of(Primer::Layout::GUTTER_OPTIONS) %>
+    # @param flow_row_until [Symbol] When the `Layout` should change from a row flow into a column flow. <%= one_of(Primer::Layout::FLOW_ROW_UNTIL_OPTIONS) %>
+    # @param sidebar_width [Symbol] <%= one_of(Primer::Layout::SIDEBAR_WIDTH_OPTIONS) %>
+    # @param sidebar_placement [Symbol] <%= one_of(Primer::Layout::SIDEBAR_PLACEMENT_OPTIONS) %>
+    # @param sidebar_flow_row_placement [Symbol] Sidebar placement when `Layout` is flowing as row. <%= one_of(Primer::Layout::SIDEBAR_FLOW_ROW_PLACEMENT_OPTIONS) %>
+    # @param main_width [Symbol] <%= one_of(Primer::Layout::MAIN_WIDTH_OPTIONS) %>
+    # @param divider [Boolean] Wether or not to add a divider between `main` and `sidebar`.
+    # @param divider_flow_row_variant [Symbol] Variants for the divider when `Layout` is flowing as row. <%= one_of(Primer::Layout::DIVIDER_FLOW_ROW_VARIANT_OPTIONS) %>
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(
+      density: DENSITY_DEFAULT,
```

Comment:
@yaili is `density` a common term in our system? Maybe this would be more conventionally named as `variant`?

Reply:
What is it relating to? If it's relating to the amount of items and space between within a screen, density would be correct.